modules/PhyVersoBSP/connector_1/evse_board_supportImpl.cpp

```diff
@@ -0,0 +1,119 @@
+
+        if (mod->config.conn1_has_socket) {
+            caps.connector_type = types::evse_board_support::Connector_type::IEC62196Type2Socket;
+        } else {
+            caps.connector_type = types::evse_board_support::Connector_type::IEC62196Type2Cable;
+        }
+    }
+
+    mod->serial.signal_cp_state.connect([this](int connector, CpState s) {
+        if (connector == 1 and s not_eq last_cp_state) {
+            publish_event(to_bsp_event(s));
+            EVLOG_info << "[1] CP State changed: " << to_bsp_event(s);
+            last_cp_state = s;
+        }
+    });
+    mod->serial.signal_set_coil_state_response.connect([this](int connector, CoilState s) {
+        if (connector == 1) {
+            EVLOG_info << "[1] Relais: " << (s.coil_state ? "ON" : "OFF");
+            publish_event(to_bsp_event(s));
+        }
+    });
+
+    mod->serial.signal_telemetry.connect([this](int connector, Telemetry t) {
+        if (connector == 1) {
+            EVLOG_info << "[1] CP Voltage: " << t.cp_voltage_hi << " " << t.cp_voltage_lo;
+        }
+    });
+}
+
+void evse_board_supportImpl::ready() {
+}
+
+void evse_board_supportImpl::handle_setup(bool& three_phases, bool& has_ventilation, std::string& country_code) {
+    // your code for cmd setup goes here
+}
+
+types::evse_board_support::HardwareCapabilities evse_board_supportImpl::handle_get_hw_capabilities() {
+    std::scoped_lock lock(caps_mutex);
+    return caps;
+}
+
+void evse_board_supportImpl::handle_enable(bool& value) {
+    if (value) {
+        mod->serial.set_pwm(1, 10000);
+    } else {
+        mod->serial.set_pwm(1, 0);
+    }
+}
+
+void evse_board_supportImpl::handle_pwm_on(double& value) {
+    if (value >= 0 && value <= 100.) {
+        mod->serial.set_pwm(1, value * 100);
+    }
+}
+
+void evse_board_supportImpl::handle_pwm_off() {
+    mod->serial.set_pwm(1, 10000);
+}
+
+void evse_board_supportImpl::handle_pwm_F() {
+    mod->serial.set_pwm(1, 0);
+}
+
+void evse_board_supportImpl::handle_allow_power_on(types::evse_board_support::PowerOnOff& value) {
+    if (mod->config.conn1_dc) {
+        mod->serial.set_coil_state_request(1, CoilType_COIL_DC1, value.allow_power_on);
+        // FIXME: implement in MCU with feedback
+        if (value.allow_power_on) {
+            publish_event({types::board_support_common::Event::PowerOn});
+        } else {
+            publish_event({types::board_support_common::Event::PowerOff});
+        }
+    } else {
+        mod->serial.set_coil_state_request(1, CoilType_COIL_AC, value.allow_power_on);
+    }
+}
+
+void evse_board_supportImpl::handle_ac_switch_three_phases_while_charging(bool& value) {
+    // your code for cmd ac_switch_three_phases_while_charging goes here
+}
+
+void evse_board_supportImpl::handle_evse_replug(int& value) {
+    // your code for cmd evse_replug goes here
+}
+
+types::board_support_common::ProximityPilot evse_board_supportImpl::handle_ac_read_pp_ampacity() {
+    // IMPLEMENT ME
```

Comment:
PP support is currently not available? Just DC support intended?

Reply:
We've [implemented](https://github.com/Qwello/everest-core/blob/78e4b42b80e472c36c432f7b82966660298b2190/modules/PhyVersoBSP/connector_1/evse_board_supportImpl.cpp#L103-L122) in qwello's fork. We will add it here.